### ISSUE TITLE
List Block styles are impacting several other blocks in editor and live site

### ISSUE THREAD

**Issue by cuemarie** [WordPress/gutenberg#50486]
### Description

There have been several reports floating around trying to describe this, but more keep arising, so I'm hoping this report can help centralize the issue a bit. In short:

The List Block's default styles and added Global Styles are also applying to other blocks in both the editor and the live site.  On the live site, the symptoms present when a post contains a List block, but the editor shows symptoms even without the list block present.

### Step-by-step reproduction instructions

1. Activate TT3 and set the default home template
2. Open Site Editor, and create a navigation menu with a submenu included
3. Open Global Styles > Blocks > List block, and apply a custom color to the TEXT and BACKGROUND fields
4. Save changes and view the site
5. Return to the editor and add a List block to the template somewhere.
6. Save changes and view the site again.

### Global Styles > Blocks > List Block > Text Color impacts blocks in editor

https://github.com/WordPress/gutenberg/assets/27249804/06dec6e5-ec24-4791-b414-e872928e5c24

Text Color - Impacted blocks in editor:
- Post Featured Image (overlay)
- Post Excerpt

Text Color - Impacted blocks in live site:
- None

---

### Adding List Block to template brings those symptoms and more to the live site

https://github.com/WordPress/gutenberg/assets/27249804/4ffed816-dad0-4fdc-a543-b5ab792be506

Text Color - Impacted blocks in editor:
- Post Featured Image (overlay)
- Post Excerpt

Text Color - Impacted blocks in live site:
- Navigation Block
- Post Excerpt

---

### List block impacting the padding of the submenu (https://github.com/WordPress/gutenberg/issues/49445)

https://github.com/WordPress/gutenberg/assets/27249804/ea2d9688-7f9a-40fd-ae3a-3c16d1729386

Padding - Impacted blocks in editor:
- None

Padding - Impacted blocks in live site:
- Navigation block (submenu)

---

### Adding a Background color to Global Styles > Blocks > List Block 

https://github.com/WordPress/gutenberg/assets/27249804/5ce6e444-8d1c-4a4a-aed0-ce8f26bdd9a7

BG Color - Impacted blocks in editor:
- None

BG Color - Impacted blocks in live site:
- Navigation Block

---

### Removing the List block from the template removes all symptoms from live site

https://github.com/WordPress/gutenberg/assets/27249804/1060033b-e647-416e-88d5-1ab0ccdbc06c

### Environment info

- WordPress 6.2
- Gutenberg 15.7.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by juanfra** [comment]
Thanks for sharing!

I think one is pretty much related to what's also happening with the [paragraph block styles](https://github.com/WordPress/gutenberg/issues/46863).

The list block is not using a `wp-block-*` class, affecting any other list present on the site. To solve this, we'll need to use a class for the list block and keep the modified styles just for that class (something similar to what I proposed [in my PR](https://github.com/WordPress/gutenberg/pull/47282) for the paragraph block, or [this other approach](https://github.com/WordPress/gutenberg/pull/42269)).

**Comment by jamiepalatnik** [comment]
I ran into this issue as well and ended up using custom CSS as a workaround to remove extra padding added to the Navigation block.

**Comment by donalirl** [comment]
I used the following CSS to address this, until the bug can be fixed:

```
/* unset padding on the navigation block */ 
.wp-block-navigation ul {
    padding: unset;
}
```

**Comment by simonwammerfors** [comment]
I've also run into a similar issue. In my case I've set the font-size for the navigation in theme.json. On pages with the list block present however, inline CSS belonging to that block overrides the font-size for the navigation.

**Comment by iamtakashi** [comment]
I've also run into this issue. In my case, list elements with a background colour had an unexpected style when I set paddings to the list block because of this CSS added to it. The target seems to be too broad.

```
ol.has-background,ul.has-background {
         padding: 1.25em 2.375em
}
```
![Screenshot 2023-07-25 at 10 02 50](https://github.com/WordPress/gutenberg/assets/908665/b8112ef9-a8ba-4f02-a422-5fd275023b4a)

**Steps to reproduce this.**
- Add a list block to a page/post.
- Add background colour to the list block.
- Add a navigation block and add background colour to the submenu background
- Set paddings for the List block in Global Styles
- See the page/post on the front of the site.

**Comment by iamtakashi** [comment]
Just saw another unwanted effect in the editor caused by these "too-broad" selectors from Global Styles. `.editor-styles-wrapper ol, .editor-styles-wrapper ul`

Once you change the horizontal padding for the list block, the post template that uses list item breaks. Notice the misalignment of the posts. They should remain unchanged by changing the padding change in the list block.

![Screenshot 2023-08-15 at 12 00 09](https://github.com/WordPress/gutenberg/assets/908665/23200f34-2cfc-475c-a8ed-8df9a00b60ee)

**Comment by jordesign** [comment]
Just noting that we're seeing mentions of this in the Support Forums as well.

* https://wordpress.org/support/topic/list-block-affecting-navigation-block-size/?view=all
* https://wordpress.org/support/topic/word-wrap-on-drop-down-menus/
* https://wordpress.org/support/topic/navigation-block-not-responding-consistently-in-different-posts/

**Comment by aleone89** [comment]
Another report of this here: https://wordpress.com/forums/topic/navigation-menu-appearance-varies/?view=all#post-3989938

**Comment by annezazu** [comment]
@ellatrix flagging this for you.

**Comment by ramonjd** [comment]
Is @juanfra's [answer](https://github.com/WordPress/gutenberg/issues/50486#issuecomment-1543872495) the correct one? I can't think of another way to isolate the list block from other list elements used about the place.

Unless Gutenberg stops using list elements in the post template block and others 😄 

_Edit_ I'm testing something similar to #50486 over at https://github.com/WordPress/gutenberg/pull/54896

**Comment by adamwoodnz** [comment]
This caused issues on the [Showcase](https://wordpress.org/showcase/) site, and [we worked around it by removing the background from the navigation block](https://github.com/WordPress/wporg-showcase-2022/commit/8287278a08ee38b3c400285b5f21b508b61b8a41).

**Comment by talldan** [comment]
This issue seems to be a duplicate of the much older issue #12420. That one has a PR associated with it (#56469).

**Comment by jordesign** [comment]
Another report in the forums:
https://wordpress.org/support/topic/possible-conflict-between-list-block-and-submenu-formatting/?view=all

**Comment by annezazu** [comment]
[After a review](https://wordpress.slack.com/archives/C02QB2JS7/p1707152945370159) by core editor triage leads and core editor tech leads, this has been removed from the board for 6.5 consideration.

**Comment by colorful-tones** [comment]
> [After a review](https://wordpress.slack.com/archives/C02QB2JS7/p1707152945370159) by core editor triage leads and core editor tech leads, this has been removed from the board for 6.5 consideration.

@annezazu did we forget to remove this from the board, or was there a reason we did not, and do you mind providing it please?

**Comment by annezazu** [comment]
Hah! This should have been removed. I bet I was moving too fast. Done.

**Comment by ellatrix** [comment]
Not sure what I can do here. This was introduced in #21387 by @ZebulanStanphill. @ZebulanStanphill, could we change this to use a class? I think it's not right that we're using a tag selector only.

**Comment by xtremeranger** [comment]
Now that the list has a class name (#56469) can the style.scss be edited to fix the issue mentioned in @iamtakashi [comment](https://github.com/WordPress/gutenberg/issues/50486#issuecomment-1649448957)?

EDIT: I spoke too soon, sorry. The change to lower the specificity has already been put in place.

**Comment by ellatrix** [comment]
@carolinan Was this fixed in #56469?

**Comment by carolinan** [comment]
yes, thank you for the reminder. I can confirm that this is solved. (I am using WP 6.6-RC3-58711)
